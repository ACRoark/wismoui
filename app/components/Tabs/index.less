@import (reference) '../../styles/themes/index.less';

@tabs-cls-prefix: ~'@{dte-prefix}-tabs';

.@{tabs-cls-prefix} {
  background-color: @dte-gray-300;
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 70px;
  width: 100%;

  @media screen and (min-width: 325px) {
    padding: 0px;
  }
  @media screen and (min-width: 600px) {
    padding: 0px 60px;
  }
  @media screen and (min-width: 800px) {
    padding: 0px 93px;
  }
  @media screen and (min-width: 1024px) {
    padding: 0px 138px;
  }
  @media screen and (min-width: 1280px) {
    padding: 0px 200px;
  }
  @media screen and (min-width: 1440px) {
    padding: 0px 245px;
  }
  @media screen and (min-width: 1680px) {
    padding: 0px 320px;
  }
  @media screen and (min-width: 1920px) {
    padding: 0px 405px;
  }

  &-left {
    margin-right: 10px;
  }

  &-selected {
    align-items: center;
    background-color: @dte-white;
    border-top: 4px solid @dte-blue;
    color: @dte-blue;
    display: flex;
    height: 60px;
    justify-content: center;
    margin-top: 10px;
    padding-bottom: 4px;
    width: 50%;
  }

  &-unselected {
    align-items: center;
    background-color: @dte-gray-200;
    color: @dte-gray-500;
    cursor: pointer;
    display: flex;
    height: 60px;
    justify-content: center;
    margin-top: 10px;
    width: 50%;
  }
}
