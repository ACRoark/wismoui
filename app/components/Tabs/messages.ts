import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.tabs';

export default defineMessages({
  startService: {
    id: `${scope}.start-service`,
  },
  stopService: {
    id: `${scope}.stop-service`,
  },
});
