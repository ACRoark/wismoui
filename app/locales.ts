export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
