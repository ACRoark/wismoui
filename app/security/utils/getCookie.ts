const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
