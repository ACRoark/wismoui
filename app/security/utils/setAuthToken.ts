import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
