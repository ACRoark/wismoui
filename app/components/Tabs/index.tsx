import React, { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';

import messages from './messages';

import './index.less';

interface ITabsProps {
  selectionChanged: (str: string) => void;
}

const Tabs: React.FC<ITabsProps> = (props: ITabsProps): React.ReactElement => {
  const { selectionChanged } = props;
  const [tab, setTab] = useState('start');

  const handleStart = (): void => {
    setTab('start');
  };

  const handleStop = (): void => {
    setTab('stop');
  };

  const start = tab === 'start' ? 'selected' : 'unselected';
  const stop = tab === 'stop' ? 'selected' : 'unselected';

  useEffect((): void => {
    selectionChanged(tab);
  }, [tab]);

  return (
    <>
      <div className="dte-wismo-tabs">
        <div className={`dte-wismo-tabs-left dte-wismo-tabs-${start} `} onClick={handleStart}>
          <FormattedMessage {...messages.startService} />
        </div>
        <div className={`dte-wismo-tabs-${stop}`} onClick={handleStop}>
          <FormattedMessage {...messages.stopService} />
        </div>
      </div>
    </>
  );
};

export default Tabs;
