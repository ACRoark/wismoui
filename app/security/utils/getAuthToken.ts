import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
