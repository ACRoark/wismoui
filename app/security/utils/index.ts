import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
