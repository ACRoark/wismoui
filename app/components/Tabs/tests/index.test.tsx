import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import Tabs from '..';

const selectionChanged = jest.fn();

describe('<Tabs />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<Tabs selectionChanged={selectionChanged} />, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should render correctly when the current language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(<Tabs selectionChanged={selectionChanged} />, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
