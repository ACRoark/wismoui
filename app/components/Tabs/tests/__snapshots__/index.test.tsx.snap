// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render correctly when the current language is en 1`] = `
<div
  className="dte-wismo-tabs"
>
  <div
    className="dte-wismo-tabs-left dte-wismo-tabs-selected "
    onClick={[Function]}
  >
    Start Service
  </div>
  <div
    className="dte-wismo-tabs-unselected"
    onClick={[Function]}
  >
    Stop Service
  </div>
</div>
`;

exports[`<Tabs /> should render correctly when the current language is es 1`] = `
<div
  className="dte-wismo-tabs"
>
  <div
    className="dte-wismo-tabs-left dte-wismo-tabs-selected "
    onClick={[Function]}
  >
    Comience el servicio
  </div>
  <div
    className="dte-wismo-tabs-unselected"
    onClick={[Function]}
  >
    Parada de servivio
  </div>
</div>
`;
